src/lib/holocene/api-pagination.svelte

```diff
@@ -30,10 +30,21 @@
     if (error) error = undefined;
   }
 
+  $: isEmpty = $store.visibleItems.length === 0 && !$store.loading;
+  $: pageSizeChange =
```

Comment:
Running into a few things:
![Screenshot 2023-01-09 at 10 29 51 AM](https://user-images.githubusercontent.com/15069288/211370487-c927b39c-d1c7-4170-a65c-c093a31949a3.png)

1. On a running workflow the number of `visibleItems` is increasing, but the total remains the same and it gets into a weird state such as in the screenshot above where there are 19 events, but it still says the total is 13.
2. If I pass in `pageSizeOptions` and change the number it will trigger a refetch, but it always just fetches **all** of the events. Do we want to update `fetchPaginatedEvents` to account for a page size? I realize it's probably fine to just have the default 100 for now, but not sure if that will change in the future.
3. There is an unused `supportsReverseOrder` import in `events-service.ts`.

Reply:
Great finds!

1. Because we have to fetch the start and end events for input/result/total event count plus the first page of events, there can be a mismatch on the total count due to timing. I.e start/end call resolves with 13 events, first page resolves with 19 events. I'll see if we can do something about that, but in a separate PR.
2. Because the event history can never guarantee exactly how many events can be returned (it's batched), we don't use a pageSize option in getPaginatedEvents (_pageSize = 100 is there to appease the TS gods, but it's not actually used in the requestFromAPI).
3. Removed

Reply:
> Because the event history can never guarantee exactly how many events can be returned 

Ah I forgot about this fun quirk 🤦‍♀️